### PR TITLE
Support taxonomy collection args on posts endpoint.

### PR DESCRIPTION
**Pull request by joehoyle**
See #2277

### CODE REVIEW

File: lib/endpoints/class-wp-rest-posts-controller.php [hunk 1]
```diff
@@ -1695,6 +1709,18 @@ public function get_collection_params() {
 		$params['filter'] = array(
 			'description'       => __( 'Use WP Query arguments to modify the response; private query vars require appropriate authorization.' ),
 		);
+
+		$taxonomies = wp_list_filter( get_object_taxonomies( $this->post_type, 'objects' ), array( 'show_in_rest' => true ) );
+		foreach ( $taxonomies as $taxonomy ) {
+			$base = ! empty( $taxonomy->rest_base ) ? $taxonomy->rest_base : $taxonomy->name;
+
+			$params[ $base ] = array(
+				'description'       => sprintf( _( 'Limit result set to all items that have the specified term assigned in the %s taxonomy.' ), $base ),
```
Comment (danielbachhuber) [inline lib/endpoints/class-wp-rest-posts-controller.php]: `__()` instead of `_()`

**Comment by danielbachhuber** [comment]
Should we error if an invalid term id is provided?

**Comment by joehoyle** [comment]
I think not, all the checking for a large amount of terms would be slow IMO.

**Comment by danielbachhuber** [comment]
Ok, just one nit then

**Comment by joehoyle** [comment]
:+1: fixed in 6c1a4d1

**Comment by kadamwhite** [comment]
This is just by ID, correct—so `tags=tagslug` wouldn't work? Fine if so, just making sure I understand the code

**Comment by kadamwhite** [comment]
> Should we error if an invalid term id is provided?

I would expect to just get an empty collection back

**Comment by kadamwhite** [comment]
Final Q: does this give any ability to query by multiple categories?

**Comment by danielbachhuber** [comment]
> This is just by ID, correct

Correct

> Final Q: does this give any ability to query by multiple categories?

Yes

**Comment by danielbachhuber** [comment]
> Final Q: does this give any ability to query by multiple categories?

Although, I believe this is `AND` and not `OR`

**Comment by BE-Webdesign** [comment]
When the filter[tax_query] is in you can specify that in addition to get either `AND` or `OR`. I also believe the default is OR but could be wrong don't remember. https://github.com/WP-API/WP-API/pull/2273

**Comment by kadamwhite** [comment]
> > Final Q: does this give any ability to query by multiple categories?
> 
> Yes

Can you demonstrate what that query would look like?

**Pull request merged**